xmpp.go

```diff
@@ -315,19 +327,16 @@ func (c *Client) init(o *Options) error {
 
 		mechanism := ""
 		for _, m := range f.Mechanisms.Mechanism {
-			if m == "ANONYMOUS" {
```

Comment:
ANONYMOUS is already handled above when there is no user and password set.  This should be the correct and distinct way to connect.